volume/ui.go

```diff
@@ -40,21 +38,34 @@ func RenderVolumesList(volumes []types.Volume) {
 		},
 	}
 
-	rows := make([]ui.Row, len(volumes))
+	rows := make([]ui.Row, 0, len(volumes))
 
 	if len(volumes) == 0 {
 		ui.Infof("No existing volumes")
 		return
 	}
 
-	for idx, volume := range volumes {
-		rows[idx] = []string{
-			volume.ID,
+	if highlight != nil {
+		rows = append(rows, []string{
+			highlight.Name + highlighted,
+			fmt.Sprintf("%d GB", highlight.Size),
+			highlight.State.CreatedAt.Format(time.RFC3339),
+			highlight.Provider.DisplayName(),
+		})
+	}
+
+	for _, volume := range volumes {
+		if highlight != nil {
+			if volume.ID == highlight.ID {
+				continue
+			}
+		}
+		rows = append(rows, []string{
 			volume.Name,
 			fmt.Sprintf("%d GB", volume.Size),
 			volume.State.CreatedAt.Format(time.RFC3339),
 			volume.Provider.DisplayName(),
-		}
+		})
```

Comment:
wouldn't it be much simpler to do all of this in one loop? You jut need one additional if statement:

```
if highlight != nil &&  volume.ID == highlight.ID {
rows[idx] = []string{
			highlight.Name + highlighted,
			fmt.Sprintf("%d GB", highlight.Size),
			highlight.State.CreatedAt.Format(time.RFC3339),
			highlight.Provider.DisplayName(),
		}
continue
}
```

Reply:
No because that would result in there being an extra entry in the list

Reply:
You would need additional handling for the case where the highlight is not nil to not insert it twice, in general I don't think trying to do everything in fori loops is simpler than appending, in the same vein that fori is notably more complex than foreach.